### PR TITLE
net: shell: Add support to set Ethernet MAC address

### PR DESCRIPTION
**Pull request by krish2718**
Add a shell command to set Ethernet MAC address, useful in configuring custom/local MAC addresses.

Signed-off-by: Krishna T <krishna.t@nordicsemi.no>

### CODE REVIEW

**Comment by krish2718** [comment]
@sachinthegreen @sr1dh4r FYI

File: subsys/net/ip/net_shell.c [hunk 1]
```diff
@@ -3563,6 +3566,51 @@ static int cmd_net_iface(const struct shell *shell, size_t argc, char *argv[])
...
+		PR_WARNING("MAC address can be set only for Ethernet\n");
+		return -ENOEXEC;
+	}
+
+	if (net_bytes_from_str(mac_addr.addr, sizeof(mac_addr), argv[2]) < 0) {
+		PR_WARNING("Invalid MAC address\n");
+		return -ENOEXEC;
+	}
+
+	ret = net_if_set_link_addr(iface, mac_addr.addr, sizeof(mac_addr), NET_LINK_ETHERNET);
```
Comment (jukkar) [inline subsys/net/ip/net_shell.c:3604]: The system will disallow changing of the MAC address if the interface is running. Perhaps we could check the error code here (-EPERM) and print a better message if that happens.

**Review by jukkar — DISMISSED** [review]
LGTM

**Review by rlubos — DISMISSED** [review]

**Comment by krish2718** [comment]
* Added a new commit to fix duplicate tag (struct shell *shell) everywhere
* Fixed few issues: Defining ret, missing command in command definition
* Used `goto err` instead of returning from all negative paths

**Review by rlubos — DISMISSED** [review]

**Comment by rlubos** [comment]
Something's off with [97ac193](https://github.com/zephyrproject-rtos/zephyr/pull/53539/commits/97ac1938157c89c3c195e8ebfaad816ebc6f1175), the git commit author name is lowercase (`krishna T`), it seems to make CI unhappy.

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**